lib/wireprotocol/3_2_support.js

```diff
@@ -9,41 +9,34 @@ const BSON = retrieveBSON();
 const Long = BSON.Long;
 const ReadPreference = require('../topologies/read_preference');
 const TxnState = require('../transactions').TxnState;
-const applyCommonQueryOptions = require('./shared').applyCommonQueryOptions;
 const isMongos = require('./shared').isMongos;
+const databaseNamespace = require('./shared').databaseNamespace;
+const collectionNamespace = require('./shared').collectionNamespace;
 
 class WireProtocol {
   insert(server, ns, ops, options, callback) {
-    executeWrite(server, 'insert', 'documents', ns, ops, options, callback);
+    executeWrite(this, server, 'insert', 'documents', ns, ops, options, callback);
   }
 
   update(server, ns, ops, options, callback) {
-    executeWrite(server, 'update', 'updates', ns, ops, options, callback);
+    executeWrite(this, server, 'update', 'updates', ns, ops, options, callback);
   }
 
   remove(server, ns, ops, options, callback) {
-    executeWrite(server, 'delete', 'deletes', ns, ops, options, callback);
+    executeWrite(this, server, 'delete', 'deletes', ns, ops, options, callback);
   }
 
   killCursor(server, ns, cursorState, callback) {
-    const bson = server.s.bson;
-    const pool = server.s.pool;
-    const parts = ns.split(/\./);
-    const commandns = `${parts.shift()}.$cmd`;
     const cursorId = cursorState.cursorId;
-    const killcursorCmd = {
-      killCursors: parts.join('.'),
+    const killCursorCmd = {
+      killCursors: collectionNamespace(ns),
       cursors: [cursorId]
     };
 
-    const query = new Query(bson, commandns, killcursorCmd, {
-      numberToSkip: 0,
-      numberToReturn: -1,
-      checkKeys: false,
-      returnFieldSelector: null
-    });
+    const options = {};
+    if (typeof cursorState.session === 'object') options.session = cursorState.session;
 
-    function killCursorCallback(err, result) {
+    this.command(server, ns, killCursorCmd, options, (err, result) => {
```

Comment:
but we'd be needlessly creating a no-op function for every call. If DRYness is your goal, we might be better off using something like the good old `handleCallback`: which just this type check for us, but also adds an extra layer to stack traces

Reply:
If we are worried about creating the callback lots of times, we can just do `const noOp = () => {};` at the top of the file and reuse it.

Reply:
Okay, I changed this. But really, we are the callers here and shouldn't ever call without a callback. In the future, I suggest we solve this at the call site rather than programming defensively against ourselves.